### PR TITLE
microscheme: refactor, fix build on darwin and cross-compilation

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Test with
```ShellSession
$ nix build .#pkgsCross.raspberryPi.microscheme
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113369` at 8232dbff run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/eb0bbaa5fd329699a16694774c80a161">microscheme</a></li></li>
  </ul>
</details>

**Comment by siraben** [comment]
Case-sensitive filenames strike again!

File: pkgs/development/compilers/microscheme/default.nix [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
   version = "0.9.3";
 
-  src = fetchzip {
-    name = "${pname}-${version}-src";
-    url = "https://github.com/ryansuchocki/microscheme/archive/v${version}.tar.gz";
-    sha256 = "1r3ng4pw1s9yy1h5rafra1rq19d3vmb5pzbpcz1913wz22qdd976";
+  src = fetchFromGitHub {
+    owner = "ryansuchocki";
+    repo = "microscheme";
+    rev = "b80901f4c7f42a372d51d92fe501263f6f54fd1a";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/microscheme/default.nix:10]: Does upstream not tag releases?

File: pkgs/development/compilers/microscheme/default.nix [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
 
-  buildInputs = [ makeWrapper vim ];
-
-  installPhase = ''
-    make install PREFIX=$out
+  postPatch = ''
+    substituteInPlace makefile --replace gcc ${stdenv.cc.targetPrefix}cc
   '';
 
+  nativeBuildInputs = [ makeWrapper vim ];
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/microscheme/default.nix:18]: You are sure about vim as a build time dependency?

File: pkgs/development/compilers/microscheme/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/development/compilers/microscheme/default.nix]: Oops, you are right.

File: pkgs/development/compilers/microscheme/default.nix (hunk 2 above)
Comment (siraben) [reply pkgs/development/compilers/microscheme/default.nix:18]: Yeah the build script calls `xxd`. Is there a more minimal way to provide this?

Comment (SuperSandro2000) [reply pkgs/development/compilers/microscheme/default.nix:18]: I think unixtools.xxd should be the right one.

Comment (siraben) [reply pkgs/development/compilers/microscheme/default.nix:18]: Yeah, https://github.com/NixOS/nixpkgs/blob/0b7f4e404c23a3f4526166d029e3c59813721fbf/pkgs/top-level/unixtools.nix#L175

**Pull request merged**